app/tests/unit/lib/theme/CustomFields.test.tsx

```diff
@@ -0,0 +1,247 @@
+      componentUnderTest.getByText("I replaced the ID")
+    ).toBeInTheDocument();
+    expect(componentUnderTest.getAllByText("ADDED")).toHaveLength(3);
+  });
+
+  it("shows data has been removed when there is old data and no new data", () => {
+    const componentUnderTest = render(
+      <FormBase
+        tagName={"dl"}
+        fields={CUSTOM_DIFF_FIELDS}
+        schema={testSchema as JSONSchema7}
+        uiSchema={uiTestSchema}
+        formData={null}
+        formContext={{
+          oldData: oldFormData,
+          oldUiSchema: oldUiTestSchema,
+          operation: "CREATE",
+        }}
+      />
+    );
+
+    expect(componentUnderTest.getByText("stringTest OLD")).toBeInTheDocument();
+    expect(componentUnderTest.getByText("$200")).toBeInTheDocument();
+    expect(
+      componentUnderTest.getByText("I replaced the OLD ID")
+    ).toBeInTheDocument();
+    expect(componentUnderTest.getAllByText("REMOVED")).toHaveLength(3);
+  });
+
+  it("shows data has been removed when operation is 'ARCHIVE'", () => {
+    const componentUnderTest = render(
+      <FormBase
+        tagName={"dl"}
+        fields={CUSTOM_DIFF_FIELDS}
+        schema={testSchema as JSONSchema7}
+        uiSchema={uiTestSchema}
+        formData={formData}
+        formContext={{
+          oldData: oldFormData,
+          oldUiSchema: oldUiTestSchema,
+          operation: "ARCHIVE",
+        }}
+      />
+    );
+
+    expect(componentUnderTest.getByText("stringTest OLD")).toBeInTheDocument();
+    expect(componentUnderTest.getByText("$200")).toBeInTheDocument();
+    expect(
+      componentUnderTest.getByText("I replaced the OLD ID")
+    ).toBeInTheDocument();
+    expect(componentUnderTest.getAllByText("REMOVED")).toHaveLength(3);
+  });
+
+  it("shows the text from 'ui:options' when 'ui:options' exists in a NumberField's uiSchema", () => {
+    const componentUnderTest = render(
+      <FormBase
+        tagName={"dl"}
+        fields={CUSTOM_DIFF_FIELDS}
+        schema={testSchema as JSONSchema7}
+        uiSchema={uiTestSchema}
+        formData={formData}
+        formContext={{
+          oldData: oldFormData,
+          oldUiSchema: oldUiTestSchema,
+          operation: "UPDATE",
+        }}
+      />
+    );
+    expect(
+      componentUnderTest.getByText("I replaced the ID")
+    ).toBeInTheDocument();
+    expect(componentUnderTest.queryByText("1")).not.toBeInTheDocument();
+  });
+
+  it("shows the number from formDatawhen 'ui:options' does not exist in a NumberField's uiSchema", () => {
+    const componentUnderTest = render(
+      <FormBase
+        tagName={"dl"}
+        fields={CUSTOM_DIFF_FIELDS}
+        schema={testSchema as JSONSchema7}
+        uiSchema={uiTestSchema}
+        formData={formData}
+        formContext={{
+          oldData: oldFormData,
+          oldUiSchema: oldUiTestSchema,
+          operation: "UPDATE",
+        }}
+      />
+    );
+    expect(componentUnderTest.getByText("$100")).toBeInTheDocument();
```

Comment:
This is still testing the `MoneyWidget` from the `ui:schema`, we probably should change the uiTestSchema to not have that set? Or maybe this test is not needed